spring-cloud-dataflow-admin-local/src/main/java/org/springframework/cloud/dataflow/admin/spi/local/OutOfProcessModuleDeployer.java

```diff
@@ -235,7 +236,19 @@ public String getId() {
 
 		@Override
 		public DeploymentState getState() {
-			return isAlive(process) ? DeploymentState.deployed : DeploymentState.failed;
+			boolean alive = isAlive(process);
+			if (!alive) {
+				return DeploymentState.failed;
+			}
+			try {
+				HttpURLConnection urlConnection = (HttpURLConnection) moduleUrl.openConnection();
+				urlConnection.connect();
+				urlConnection.disconnect();

```

Comment:
Not sure. That thing in InProcessMD was an attempt to workaround a test failure (which reminds me I need to disable some tests for now). Unless you want to have a look at them